docs/iot/guides/update-control-api.md

```diff
@@ -0,0 +1,142 @@
+```cpp
+int error_code;
+bool is_available;
+
+error_code = update_control_get_property(UPDATE_CONTROL_PROPERTY_UPDATE_AVAILABLE, (void **)&is_available);
+if (error_code == UPDATE_CONTROL_ERROR_NONE) {
+  if (is_available)
+    // New version platform is available
+    ...
+}
+```
+
+## Downloading and Updating to New Version
+
+If the newer version is available, the application can download it and update to the found one.
+The functions `update_control_download_package()` and `update_control_do_update()` are offered and their usage is similar to `update_control_get_property()`:
+```cpp
+int error_code;
+
+error_code = update_control_download_package();
+if (error_code != UPDATE_CONTROL_ERROR_NONE)
+  return;
+
+error_code = update_control_do_update();
+if (error_code != UPDATE_CONTROL_ERROR_NONE)
+  return;
+```
+Usually, downloading and updating are processed sequentially but the application also can call each function separately for intending operation.
+
+## Making an Update Reservation
+
+The device especially IoT may be critical at it's operation time. Because updating procedure may include unavailable time (ex. rebooting), it needs to schedule software update at specific time.
+In this case, the application can use `update_control_make_reservation()` function:
+```cpp
+int error_code;
+struct tm time;
+
+// Set the "time" to trigger the update
+error_code = update_control_make_reservation(&time);
+if (error_code != UPDATE_CONTROL_ERROR_NONE)
+  return;
+```
+The reserved update task also can be canceled by `update_control_cancel_reservation()`.
+
+> **Note**
+>
+> It assumes only one update reservation can be set. If the platform supports multiple reservations, they should be managed properly about cancel request.
+
+## Update Properties
+
+Besides `UPDATE_CONTROL_PROPERTY_UPDATE_AVAILABLE` mentioned at above, there are several information about update.
+They can be got through `update_control_get_property()` function and each key:
+
+| Key                                   |  Description                                  | Type (example) |
+| --------------------------------------| --------------------------------------------- |----------------|
+| `UPDATE_CONTROL_PROPERTY_NEW_VERSION` | The value of new version (ex. string, number) | string         |
+| `UPDATE_CONTROL_PROPERTY_PACKAGE_URI` | URI of update package is uploaded             | string         |
+| `UPDATE_CONTROL_PROPERTY_RESULT`      | The result of recent update procedure         | int            |
+| `UPDATE_CONTROL_PROPERTY_PACKAGE_SIZE`| The size of update package to be downloaded   | unsigned int   |
+| `UPDATE_CONTROL_PROPERTY_DESCRIPTION` | The description about found version           | string         |
+| `UPDATE_CONTROL_PROPERTY_UPDATE_AVAILABLE`| The flag of update availability           | bool           |
+
+Because their actual types are depend on platform developer, the value that will be returned for each key should be known to the application.
```

Comment:
Because their actual types are depended on the platform developer, and the value that will be returned for each key must be known to the application.

Reply:
Done.